hail/src/main/scala/is/hail/rvd/RVDType.scala

```diff
@@ -28,22 +29,25 @@ final case class RVDType(rowType: PStruct, key: IndexedSeq[String])
     .filter(i => !keySet.contains(rowType.fields(i).name))
     .toArray
 
-  val kOrd: UnsafeOrdering = kType.unsafeOrdering()
```

Comment:
making these `def`s is causing some of the usages to now create orderings for each comparison. We should create an interface here that makes it impossible for these to be created multiple times. How about:

```
  private[this] var sm: HailStageManager = _
  def setStateManager(sm: HailStateManager): Unit = this.sm = sm
  
  lazy val kOrd: UnsafeOrdering = kType.unsafeOrdering(sm)
  lazy val kInRowOrd: UnsafeOrdering =
    RVDType.selectUnsafeOrdering(rowType, kFieldIdx, rowType, kFieldIdx, sm)
  lazy val kRowOrd: UnsafeOrdering =
    RVDType.selectUnsafeOrdering(kType, Array.range(0, kType.size), rowType, kFieldIdx, sm)
```

Now you just need to call `setStateManager` somewhere above, and you're guaranteed not to create multiple orderings per type.